src/lib/gssapi/krb5/iakerb.c

```diff
@@ -569,6 +640,23 @@ iakerb_initiator_step(iakerb_ctx_id_t ctx,
 
     switch (ctx->state) {
     case IAKERB_AS_REQ:
+        if (cred->name->princ->realm.length == 0 &&
+            (cred->name->princ->realm.data == NULL ||
+             cred->name->princ->realm.data[0] == '\0')) {
```

Comment:
If `realm.length` is 0 then it is potentially undefined behavior to fetch `realm.data[0]`.  The length test alone should be sufficient.

Reply:
Fixed, thanks.